node-zerox/tests/utils.ts

```diff
@@ -0,0 +1,39 @@
+export const compareKeywords = (
+  pages: { content: string }[],
+  expectedKeywords: string[][]
+) => {
+  const keywordCounts: {
+    keywordsFound: string[];
+    keywordsMissing: string[];
+    page: number;
+    totalKeywords: number;
+  }[] = [];
+
+  for (let i = 0; i < expectedKeywords.length; i++) {
+    const page = pages[i];
+    const keywords = expectedKeywords[i];
+    const keywordsFound: string[] = [];
+    const keywordsMissing: string[] = [];
+
+    if (page && keywords) {
+      const pageContent = page.content.toLowerCase();
+
+      keywords.forEach((keyword) => {
+        if (pageContent.includes(keyword.toLowerCase())) {
```

Comment:
@tylermaran Looking for the keyword in all of the content could yield incorrect results in the following cases due to partial matches:
Keyword `2.0` matching with string "10**2.0**24", or
Keyword `conscious` matching with string "subconscious".

This is why the previous approach of splitting the content into an array, converting it into a set, and using `.has` to check for existence of the keyword would be better as it'll perform complete matches.